pkg/rest/request/request.go

```diff
@@ -54,6 +54,10 @@ func (c Client) Send(ctx context.Context, route, method string, body, responsePt
 		req.Header[key] = values
 	}
 
```

Comment:
This is a comment on a random section of the `request.go` file

i think it is a nice opportunity to try to add a test case in the `request_test.go`, meaning that it would be nice to check if the request sent to the server will receive Header `Host` with the correct value.

the function to test is `func TestClient_Send(t *testing.T) ` and it contains the following structure

```go
	type fields struct {
		c                http.Client
		middleware       Encoder
		decoderGenerator DecoderGenerator
		auth             Authenticator
	}
	type args struct {
		ctx    context.Context
		route  string
		method string
		body   interface{}
	}
	tests := []struct {
		name       string
		fields     fields
		args       args
		wantErr    bool
		wantErrReq bool
		want       interface{}
		response   interface{}
	}
```


i also believe that the test has a handler that is being used to mock the server receiving the request
```go
handler := func(w http.ResponseWriter, r *http.Request) {
	decoder := json.NewDecoder(r.Body)
	encoder := json.NewEncoder(w)

	if r.Method != tt.args.method {
		w.WriteHeader(http.StatusBadRequest)
		encoder.Encode(ierrors.New(
			"methods are not equal",
		).BadRequest())
		return
	}
	if r.URL.Path != tt.args.route {
		w.WriteHeader(http.StatusNotFound)
		encoder.Encode(ierrors.New(
			"paths are not equal",
		).BadRequest())
		return
	}

	// adds token to response
	w.Header().Add("Authorization", "Bearer mock_token")

	if tt.wantErrReq {
		w.WriteHeader(http.StatusBadRequest)
		encoder.Encode(ierrors.New(
			"wants error",
		).BadRequest())
		return
	}

	var body interface{}

	decoder.Decode(&body)
	if !reflect.DeepEqual(tt.args.body, body) {
		w.WriteHeader(http.StatusBadRequest)
		encoder.Encode(ierrors.New("").BadRequest())
		return
	}
	encoder.Encode(tt.want)
}
```

From the snippet above i assume that the test uses tt.args to check if the data from the request is being sent in the desired manner. If you want to ask question about it, i think via voicechat is best but it could be through here